src/config/store.ts

```diff
@@ -6,6 +6,39 @@ import defaults from './defaults';
 import { DefaultPresetList, type Preset } from '@/plugins/downloader/types';
 
 const migrations = {
+  '>=3.2.2'(store: Conf<Record<string, unknown>>) {
+    const scrobblerConfig = store.get('plugins.scrobbler') as Record<
+      string,
+      unknown
+    >;
+    if (scrobblerConfig) {
+      const oldEnabled = store.get(
+        'plugins.lastfm.enabled',
+      ) as boolean | undefined;
+
+      if (oldEnabled !== undefined) {
+        scrobblerConfig.scrobblers.lastfm.enabled = oldEnabled;
+        store.set('plugins.scrobbler', scrobblerConfig);
+      }
+
+      const lastFmKeys = [
+          'token',
+          'session_key',
+          'api_root',
+          'api_key',
+          'secret'
+      ];
+
+      for (const key of lastFmKeys) {
+        const value = store.get('plugins.lastfm.' + key) as string | undefined;
+        if (value !== undefined) {
+          scrobblerConfig.scrobblers.lastfm[key] = value;
+        }
+      }
+
+      store.set('plugins.scrobbler', scrobblerConfig);
+    }
```

Comment:
```suggestion
    const lastfmConfig = store.get('plugins.lastfm') as {
      enabled?: boolean;
      token?: string;
      session_key?: string;
      api_root?: string;
      api_key?: string;
      secret?: string;
    };
    if (lastfmConfig) {
      const scrobblerConfig = store.get(
        'plugins.scrobbler',
      ) as {
        enabled?: boolean;
        scrobblers: {
          lastfm: {
            enabled?: boolean;
            token?: string;
            session_key?: string;
            api_root?: string;
            api_key?: string;
            secret?: string;
          };
        };
      };

      if (lastfmConfig) {
        scrobblerConfig.enabled = lastfmConfig.enabled;
        scrobblerConfig.scrobblers.lastfm = lastfmConfig;
        store.set('plugins.scrobbler', scrobblerConfig);
      }
    }
```

Reply:
Thanks, I got that completely backwards.